expertsystem/reaction/conservation_rules.py

```diff
@@ -229,7 +237,7 @@ class CParityNodeInput:
     s_mag: NodeQuantumNumbers.s_magnitude = attr.ib()
 
 
-class CParityConservation(Rule):
+class CParityConservation:
```

Comment:
So here it's important to be aware that the developer is not warned  anymore (through `mypy` or other linters) within this module when a signature is incorrect. If I understand correctly, the `Protocol` is only enforced statically at the place where the rule is called, wereas with an ABC (`Rule`, previously), the signature is checked at the place of the definition.